src/wrapper/requestListener.ts

```diff
@@ -15,11 +15,16 @@ interface ObjectConfig {
 
   requestListener?: http.RequestListener;
 
+  // Fastify compatibility
+  server?: http.Server;
+
   default?: unknown;
   port?: unknown;
 }
 
-const isConfig = (data: unknown): data is FunctionConfig | ObjectConfig =>
+const isConfig = (
+  data: unknown,
+): data is Promise<FunctionConfig> | Promise<ObjectConfig> =>
```

Comment:
🤔 Why are these promises?

Reply:
Fastify is a confusing hybrid of callbacks and promises. To support both styles we gotta make it a Promise.

https://www.fastify.io/docs/latest/Guides/Getting-Started/#your-first-server

Reply:
<img width="1200" alt="image" src="https://user-images.githubusercontent.com/18017094/216801513-3a58aa96-46b1-4cff-9bad-86b27d79789f.png">

Reply:
I don't think this guard is properly checking for a nested `Promise` type though. The input would have to be awaited before being passed in

Reply:
https://github.com/samchungy/test-fastify-template/pull/1 if you wanna peek. Still very much a WIP. Putting down the tools for the day :)

Reply:
The app.ts calls createApp() and exports it so it should be okay

Reply:
Alright added some tests and better typings